tests/tests_unit/test_api/test_raw.py

```diff
@@ -322,6 +323,49 @@ def test_raw_row__direct_column_access():
         del row["wrong-key"]
 
 
+@pytest.mark.dsl
+def test_insert_dataframe_raises_on_duplicated_cols(cognite_client):
+    import pandas as pd
+
+    df = pd.DataFrame(
+        {
+            "a": [1, 2, 3],
+            "b": [1, 2, 3],
+            "c": [10, 20, 30],
+            "d": [10, 20, 30],
+            "e": [100, 200, 300],
+            "f": [100, 200, 300],
+        }
+    )
+    df.columns = ["a", "b", "a", "c", "a", "b"]
```

Comment:
Why not set columns above with the data?

Reply:
Notice the duplicates 😉